src/openforms/formio/formatters/default.py

```diff
@@ -0,0 +1,152 @@
+from datetime import datetime
+from typing import Any, Dict, List, Union
+
+from django.template.defaultfilters import date as fmt_date, time as fmt_time, yesno
+from django.utils.dateparse import parse_date, parse_time
+from django.utils.translation import gettext_lazy as _
+
+from openforms.plugins.plugin import AbstractBasePlugin
+from openforms.submissions.models import _join_mapped
+
+from .registry import register
+
+
+def _get_value_label(possible_values: list, value: str) -> str:
+    for possible_value in possible_values:
+        if possible_value["value"] == value:
+            return possible_value["label"]
+
+    assert isinstance(value, str), "Expected value to be a str"
+
+    return value
+
+
+class FormioFormatter(AbstractBasePlugin):
+    def __call__(
+        self, component: dict, value: Any, multiple: bool = False
+    ) -> Union[str, List]:
+        if not value:
```

Comment:
Should this explicitly check for `None`? Could `value` possibly be an empty list as well?

Reply:
I'm not sure - this should become clear by having all components in all variants and leaving stuff blank/filled out. Let's hold off on that and see what we uncover with testing.